Sources/Moya/Endpoint.swift

```diff
@@ -99,7 +72,45 @@ extension Endpoint {
         request.httpMethod = method.rawValue
         request.allHTTPHeaderFields = httpHeaderFields
 
-        return try? parameterEncoding.encode(request, with: parameters)
+        switch task {
+        case .requestPlain, .uploadFile, .uploadMultipart, .downloadDestination:
+            break
+
+        case .requestData(let data):
+            request.httpBody = data
+
+        case let .requestParameters(parameters: parameters, encoding: parameterEncoding):
+            do {
```

Comment:
Good catch, updated the cases `requestParameters` and `downloadParameters` where this was possible. This didn't work in the composite cases though as they need to to multiple things with the request.